### PR TITLE
chore(demo): Remove old demo route and mark remaining code as legacy

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Following #12990, we now only have two places where the old demo data code is used: the `/demo` route and the `setup_dev` command.

## Changes

This removes the `/demo` route and moves the remaining old code to a `legacy/` subdirectory.

### CODE REVIEW

File: posthog/management/commands/setup_dev.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from django.core.management.base import BaseCommand
 from django.db import transaction
 
-from posthog.demo import ORGANIZATION_NAME, TEAM_NAME, create_demo_data
+from posthog.demo.legacy import ORGANIZATION_NAME, TEAM_NAME, create_demo_data
```
Comment (pauldambra) [inline posthog/management/commands/setup_dev.py:4]: Should we output that setup_dev is deprecated and people should use generate_demo_data?

**Review by pauldambra — APPROVED** [review]
One thought that maybe makes it easier to delete this completely in future but otherwise 🛡️

File: posthog/management/commands/setup_dev.py (hunk 1 above)
Comment (pauldambra) [reply posthog/management/commands/setup_dev.py:4]: (GitHub wouldn't let me comment on a more sensible line in the file 🤷 )

Comment (Twixes) [reply posthog/management/commands/setup_dev.py:4]: 👍 Good idea

**Comment by Twixes** [comment]
I'll keep the Hogflix page, because it's still used by a Cypress test (plus I saw Yakko use it recently 😅), but opening it will no longer create a demo project.

**Pull request merged**